clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -2006,7 +2027,7 @@ scheduler_tmpl_t<RecordType, ReaderType>::next_record(output_ordinal_t output,
         } else if (options_.mapping == MAP_TO_ANY_OUTPUT) {
             trace_marker_type_t marker_type;
             uintptr_t marker_value;
-            if (input->processing_blocking_syscall) {
+            if (input->processing_syscall || input->processing_blocking_syscall) {
```

Comment:
Enough to check just `processing_syscall`? L2064 would set it for also maybe-blocking ones right?

Reply:
Some synthetic sequences in tests only set the maybe-blocking, and it looked painful to change them.  Added a comment.